### PR TITLE
[Merged by Bors] - refactor of NumberTheory.NumberField.Embeddings

### PR DESCRIPTION
**Pull request by xroblot**
Remove unnecessary or redundant results and golf some proofs. 

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by riccardobrasca** [comment]
I am maybe confused (the diff is a little hard to follow), but why removing `place_conjugate` (just to give an example)? Is it a special case of something more general?

**Comment by xroblot** [comment]
> I am maybe confused (the diff is a little hard to follow), but why removing `place_conjugate` (just to give an example)? Is it a special case of something more general?

Since this file became a big mess, especially everything concerning real embeddings, I decided to keep only what was needed for the proof of Dirichlet's theorem. It is possible that I went a bit too far. I'll see if I should put back some results that I deleted and I'll write a better PR description.

**Comment by riccardobrasca** [comment]
Thanks!

bors merge
